persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -678,6 +686,16 @@ sqlUnique (UniqueDef _ cname cols _) = T.concat
     , ")"
     ]
 
+schemaNamePair :: EntityDef -> SchemaEntityName
+schemaNamePair ent = schemaNamePair' (getEntityDBSchema ent) (getEntityDBName ent)
+
+schemaNamePair' :: Maybe Text -> EntityNameDB -> SchemaEntityName
+schemaNamePair' mbSchema entName = case mbSchema of
+    Nothing -> ("", entName)
+    Just "main" -> ("", entName)
+    Just "temp" -> ("temp", entName)
+    Just schema -> ("", EntityNameDB $ (schema <> "_") <> unEntityNameDB entName)
```

Comment:
Intriguing - so sqlite only has two schema, `main` and `temp`? And we mimic that by using `schema_table_name`? I'd be inclined to use `__` (two underscore) to make the distinction more clear - that way you can see `foo__bar_baz` and know that `foo` is the supposed schema, rather than `foo_bar_baz` potentially being either `BarBaz schema=foo` or `FooBarBaz` (no schema).